qiskit/tools/visualization/_state_visualization.py

```diff
@@ -5,59 +5,89 @@
 
 """
 Visualization functions for quantum states.
 """
 
+import warnings
 from functools import reduce
-
 import numpy as np
-from matplotlib import cm
-from matplotlib import pyplot as plt
-from matplotlib.patches import FancyArrowPatch
-from mpl_toolkits.mplot3d import proj3d
 from scipy import linalg
-
 from qiskit.quantum_info import pauli_group, Pauli
-from qiskit.tools.visualization import VisualizationError
-from qiskit.tools.visualization._bloch import Bloch
+from ._matplotlib import HAS_MATPLOTLIB
+
+if HAS_MATPLOTLIB:
+    from matplotlib import cm
+    from matplotlib.ticker import MaxNLocator
+    from matplotlib import pyplot as plt
+    from matplotlib.patches import FancyArrowPatch
+    from mpl_toolkits.mplot3d import proj3d
+    from qiskit.tools.visualization._error import VisualizationError
+    from qiskit.tools.visualization._bloch import Bloch
 
+if HAS_MATPLOTLIB:
+    class Arrow3D(FancyArrowPatch):
+        """Standard 3D arrow."""
 
-class Arrow3D(FancyArrowPatch):
-    """Standard 3D arrow."""
+        def __init__(self, xs, ys, zs, *args, **kwargs):
+            """Create arrow."""
+            FancyArrowPatch.__init__(self, (0, 0), (0, 0), *args, **kwargs)
+            self._verts3d = xs, ys, zs
 
-    def __init__(self, xs, ys, zs, *args, **kwargs):
-        """Create arrow."""
-        FancyArrowPatch.__init__(self, (0, 0), (0, 0), *args, **kwargs)
-        self._verts3d = xs, ys, zs
+        def draw(self, renderer):
+            """Draw the arrow."""
+            xs3d, ys3d, zs3d = self._verts3d
+            xs, ys, _ = proj3d.proj_transform(xs3d, ys3d, zs3d, renderer.M)
+            self.set_positions((xs[0], ys[0]), (xs[1], ys[1]))
+            FancyArrowPatch.draw(self, renderer)
 
-    def draw(self, renderer):
-        """Draw the arrow."""
-        xs3d, ys3d, zs3d = self._verts3d
-        xs, ys, _ = proj3d.proj_transform(xs3d, ys3d, zs3d, renderer.M)
-        self.set_positions((xs[0], ys[0]), (xs[1], ys[1]))
-        FancyArrowPatch.draw(self, renderer)
 
+def _validate_input_state(quantum_state):
+    """Validates the input to state visualization functions.
 
-def plot_hinton(rho, title='', filename=None, show=False):
+    Args:
+        quantum_state (ndarray): Input state / density matrix.
+    Returns:
+        rho: A 2d numpy array for the density matrix.
+    Raises:
+        VisualizationError: Invalid input.
+    """
+    rho = np.asarray(quantum_state)
+    if rho.ndim == 1:
+        rho = np.outer(rho, np.conj(rho))
+    # Check the shape of the input is a square matrix
+    shape = np.shape(rho)
+    if len(shape) != 2 or shape[0] != shape[1]:
+        raise VisualizationError("Input is not a valid quantum state.")
+    # Check state is an n-qubit state
+    num = int(np.log2(len(rho)))
+    if 2 ** num != len(rho):
+        raise VisualizationError("Input is not a multi-qubit quantum state.")
+    return rho
+
+
+def plot_hinton(rho, title='', figsize=None):
```

Comment:
We probably should update the function name here to be `plot_state_hinton()` to be consistent with the other functions

Reply:
done.